.circleci/config.yml

```diff
@@ -1,67 +1,91 @@
+      command-run:
+        type: string
     docker:
-      - image: python:3.6
+      - image: cimg/python:3.9
+        #auth:
+        #  username: $DOCKERHUB_USERNAME
+        #  password: $DOCKERHUB_PASSWORD
     steps:
       - checkout
       - run:
-          name: Build source distribution and install package from it
-          working_directory: ~/project/
-          # Ensure we can build a source distribution that can correctly install.
-          # "python setup.py sdist" creates dist/wikipron-x.y.z.tar.gz
-          command: |
-              pip install --progress-bar off --upgrade pip setuptools
-              python setup.py sdist
-              pip install dist/`ls dist/ | grep .tar.gz`
-      - run:
-          name: Install the full development requirements
-          working_directory: ~/project/
-          command: pip install --progress-bar off -r requirements.txt
-      - run:
-          name: Show installed Python packages
-          command: pip list
+          name: Create custom requirements
+          command: grep << parameters.command-name >> requirements.txt > << parameters.command-name >>_requirements.txt
+      - python/install-packages:
+          pkg-manager: pip
+          pip-dependency-file: << parameters.command-name >>_requirements.txt
+          cache-version: << parameters.command-name >>-v1
       - run:
-          name: Type checking
           working_directory: ~/
-          command: mypy --ignore-missing-imports project/wikipron project/tests project/data
-      - run:
-          name: Lint
-          working_directory: ~/
-          # Avoid being able to import wikipron by relative import.
-          # Test code by importing the *installed* wikipron in site-packages.
-          command: flake8 project/setup.py project/wikipron project/tests
+          command: << parameters.command-run >>
+
+  build-python:
+    parameters:
+      python-version:
+        type: string
+    docker:
+      - image: cimg/python:<< parameters.python-version >>
+        #auth:
+        #  username: $DOCKERHUB_USERNAME
+        #  password: $DOCKERHUB_PASSWORD
+    steps:
+      - checkout
+      - python/install-packages:
+          pkg-manager: pip
+          pre-install-steps:
+              - run:
+                  name: Build source distribution and install package from it
+                  command: |
+                      pip install --progress-bar off --upgrade pip setuptools
+                      python setup.py sdist
+                      pip install dist/*.tar.gz
       - run:
-          name: Check styling consistency with black
-          working_directory: ~/
-          # Avoid being able to import wikipron by relative import.
-          # Test code by importing the *installed* wikipron in site-packages.
-          command: black --line-length=79 --check project/setup.py project/wikipron project/tests project/data
+          name: Show installed Python packages
+          command: pip list -v
```

Comment:
The `-v` flag to be more verbose for where things are is nice. Thanks for adding it!